src/foam/core/FObject.js

```diff
@@ -985,7 +985,13 @@ foam.CLASS({
           // for each object since they are of the exact same
           // type.
           if ( o.hasOwnProperty(name) || props[i].factory ) {
-            this[name] = o[name];
+            let handledByPropertyClass = false;
```

Comment:
This is an odd control-structure. Why not just:

if ( ! props[i].copyValueFrom || ! iprops[i].copyValueFrom(this, o) ) this[name] = o[name];

Reply:
Oh neat. Usually I avoid that in case someone assumes the function has no side effect, but in this case it makes it look so much cleaner. Thanks! (fixed)